source/row_selection_table_model.ts

```diff
@@ -245,56 +283,64 @@ export class RowSelector {
 
   private s7() {
     this.state = 7;
-    this.clearTrueRows();
-    this.hilightedRow = this.currentRow;
-    this.selectedRows.set(this.hilightedRow, 0, true);
+    this.clearRows();
+    this.highlightedRow = this.currentRow;
+    this.selectedRows.set(this.highlightedRow, 0, true);
   }
 
+  ///??????
   private handleOperations(newOperations: Operation[]): void {
-    this.selectedRows.beginTransaction();
+    this.beginTransaction();
     for(const operation of newOperations) {
       if(operation instanceof AddRowOperation) {
         if(this.selectedRows.rowCount === 0) {
           this.selectedRows.addRow([true]);
-          this.hilightedRow = 0;
+          this.highlightedRow = 0;
           this.currentRow = 0;
           this.previousRow = 0;
         } else {
           this.selectedRows.addRow([false], operation.index);
         }
+        const row = new ArrayTableModel();
+        row.addRow([false]);
+        this.operations.push(new AddRowOperation(operation.index, row));
       } else if(operation instanceof MoveRowOperation) {
         this.selectedRows.moveRow(operation.source, operation.destination);
+        this.operations.push(
+          new MoveRowOperation(operation.source, operation.destination));
       }
     }
-    this.selectedRows.endTransaction();
+    this.endTransaction();
   }
 
-  private clearTrueRows() {
+  private clearRows() {
+    this.selectedRows.beginTransaction();
     for(let i = 0; i < this.selectedRows.rowCount; ++i) {
-      if(this.selectedRows.get(i,0)) {
-        this.selectedRows.set(i,0, false);
+      if(this.selectedRows.get(i, 0)) {
```

Comment:
No need for the ```if``` statement.

Reply:
I think it's important to always have a highlighted row. Otherwise bad things will happen in the shift case.

Reply:
I just mean you should delete line 319.